### DISCUSSION TITLE
Add option to leave voicemail

### DISCUSSION THREAD

**Discussion by Susurrus** [element-hq/element-meta#638] [Ideas]
#### What would you like to do?

If a user doesn't pick up or declines a call, I'd like to leave a voice message, similar to how conventional phones work.

#### Why would you like to do it?

Often when I'm calling someone, I'm already committed to using voice for whatever reason. I may not be able to use both hands to type, I may need to say something complex that would be too hard to type, etc. So having a seemless transition to a voice message from a missed/declined call would be very useful!

#### How would you like to achieve it?

I think it should work very similar to how hows POTS calls work with voicemail. After a certain number of rings, or the phone call being declined, the phone would say "XX did not pickup, you can leave a voice message instead after the beep BEEP" or "XX declined the call; you can leave a voice message instead after the beep BEEP". Once the beep happens a normal voice message would be sent that would end with a similar interface to ending a regular voice call. So the user would have to hit the red "end call" button to stop recording the voice message.

### Have you considered any alternatives?

Alternativel you can just do this manually. My suggestion is just smoothing out the manual process. You can already just end the call and then hit the voicemail button. What I like about this change is that it'd let you do it with no cognitive load, you just stay on the line and talk. Having to take the phone done, hit the end button, debate whether you really want to leave a message, then hit the record buttong is a lot of extra cognitive load.

### Are you willing to provide a PR?

No

**Comment by mnaturel** [comment]
Hi @Susurrus, thanks for your suggestion. As it impacts all the Element clients, I transferred the issue into the meta repository and converted it into a discussion.
